### PR TITLE
treehouses ap/bridge hotspotname rfc check (fixes #528)

### LINKED ISSUE #528: treehouses ap local|internet is accepting a non "essid" input

### PR DESCRIPTION
**Pull request by phamduchongan93**
- check if essid has less than 32 character
- check if there is no spaces.

### CODE REVIEW

**Comment by kjong** [comment]
Fixes #528

File: modules/ap.sh [hunk 1]
```diff
@@ -8,6 +8,11 @@ function ap {
   channels=(1 6 11)
   channel=${channels[$((RANDOM % ${#channels[@]}))]};
 
+  if [[ "${#essid}" -gt 32 && "$essid" = *" "* ]]; then
+    echo "Error: ssid must have less than 32 character and no spaces"
```
Comment (kjong) [inline modules/ap.sh]: `echo "Error: ssid must have less than or equal to 32 characters and no spaces"`

or

`echo "Error: ssid cannot have more than 32 characters and no spaces"`

Comment (kjong) [reply modules/ap.sh]: A more serious issue:
Running `./cli.sh ap internet "tree houses"` (so using an essid with a space and no password)  seems to crash things.

Comment (pattanawadee88) [reply modules/ap.sh]: > A more serious issue:
> Running `./cli.sh ap internet "tree houses"` (so using an essid with a space and no password) seems to crash things.

I got this problem also.

Comment (phamduchongan93) [reply modules/ap.sh]: I wrote a wrong if statement. :)

**Comment by LordJashin32** [comment]
Why is there no dash/underscore support? Adding it now.

**Comment by LordJashin32** [comment]
It seems @kjong request has been fixed. All tests are successful, approving this for merge. The network name is now filtered better but it's still a shame that bash freaks out when i put my network password in with special characters. I have to escape them with a \ . But i looked it up and its just bash's default behavior. Nothing we can sanitize out.

**Comment by phamduchongan93** [comment]
> It seems @kjong request has been fixed. All tests are successful, approving this for merge. The network name is now filtered better but it's still a shame that bash freaks out when i put my network password in with special characters. I have to escape them with a \ . But i looked it up and its just bash's default behavior. Nothing we can sanitize out.

thanks for updating the regex and WiFi character count. Before merging, let others testing this script so that we know that we can filter all invalid input. Feel free to change or rewrite the code as you wish.

**Comment by LordJashin32** [comment]
@phamduchongan93 My grammar tool tells me to change your "contains" to "contain".
image
My grammar isn't good either but I have this tool just for this reason.

File: modules/wifi.sh [hunk 2]
```diff
@@ -6,9 +6,24 @@ function wifi {
     echo "Error: name of the network missing"
     exit 1
   fi
+
   wifinetwork=$1
   wifipassword=$2
 
+  if [ -n "$wifinetwork" ]
```
[the remark this answers is not in the document]
Comment (phamduchongan93) [reply modules/wifi.sh]: @dogi understood! let me throw out a testing script before the meet up on Friday.

**Pull request merged**